### PR TITLE
Add `x/slices` package with `Map`, `Filter`, and `Reduce`

### LINKED ISSUE #273: Add new `slices` package with `Map`, `Filter`, `Reduce` functions
The package should live in dir `x/slices`.

- `Map` should be a copy of `Map` in the root package, except that the callback function should receive an index into the slice as well
- `Filter` should take a slice of `Node` and return a slice of `Node`, and take a callback function that returns a boolean
- `Reduce` should take a slice of `Node` and return a single `Node`, with a callback function to reduce

### PR DESCRIPTION
**Pull request by maragubot**
## Summary

- Adds new experimental `x/slices` package with generic slice manipulation functions:
  - `Map`: transforms slices with index-aware callback `func(int, T1) T2`
  - `Filter`: selects elements based on predicate `func(int, T) bool`
  - `Reduce`: accumulates slice into single value with `func(T2, T1) T2`
- Updates README to document `x/...` experimental packages and their compatibility policy
- Updates FAQ to point to `x/slices` for collection utilities

Fixes #273

## Test plan

- [x] All functions have tests covering edge cases
- [x] Testable examples with HTML usage
- [x] 100% test coverage
- [x] Linter passes

### CODE REVIEW

File: x/slices/slices_test.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+		if counts["banana"] != 2 {
+			t.Errorf("expected 2 bananas, got %v", counts["banana"])
+		}
+		if counts["orange"] != 1 {
+			t.Errorf("expected 1 orange, got %v", counts["orange"])
+		}
+	})
+}
+
+func TestCombinedOperations(t *testing.T) {
```
Comment (markuswustenberg) [inline x/slices/slices_test.go]: Remove this.

File: x/slices/slices_test.go [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+		result := slices.Map(input, func(i int, v int) string {
+			return strconv.Itoa(i) + ":" + strconv.Itoa(v)
+		})
+		expected := []string{"0:1", "1:2", "2:3"}
+		if !reflect.DeepEqual(expected, result) {
+			t.Errorf("expected %v, got %v", expected, result)
+		}
+	})
+
+	t.Run("transforms strings to their lengths plus index", func(t *testing.T) {
```
Comment (markuswustenberg) [inline x/slices/slices_test.go]: remove, duplicate

File: x/slices/slices_test.go [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+		result := slices.Filter(input, func(_ int, v int) bool {
+			return v%2 == 0
+		})
+		expected := []int{2, 4, 6}
+		if !reflect.DeepEqual(expected, result) {
+			t.Errorf("expected %v, got %v", expected, result)
+		}
+	})
+
+	t.Run("keeps strings with length >= 3", func(t *testing.T) {
```
Comment (markuswustenberg) [inline x/slices/slices_test.go]: doesn't add value

File: x/slices/slices_test.go [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+		result := slices.Filter(input, func(_ int, v int) bool {
+			return v%2 == 0
+		})
+		expected := []int{2, 4, 6, 8}
+		if !reflect.DeepEqual(expected, result) {
+			t.Errorf("expected %v, got %v", expected, result)
+		}
+	})
+
+	t.Run("filters structs by field value", func(t *testing.T) {
```
Comment (markuswustenberg) [inline x/slices/slices_test.go]: does not add more value

**Review by markuswustenberg — CHANGES_REQUESTED** [review]
There's a lot of subtests that don't add any more value with just using different types. Don't do that.
